src/app/store/undo/undo.state.ts

```diff
@@ -46,6 +52,12 @@ export class UndoState {
     return state.future.length > 0;
   }
 
+  constructor(private store: Store) {}
+
+  ngxsOnChanges(change: NgxsSimpleChange<UndoStateModel>) {
+    console.log(change.previousValue, change.currentValue);
+  }
```

Comment:
@garygrossgarten Maybe the new lifecylce `ngxsOnChanges` is usefully for undo/redo?